### PR TITLE
Find existing users in the DB on connect

### PR DESCRIPTION
**Pull request by OmarQurashi868**
- Server now checks if the auth username already exists in DB and loads its data
- If the user is banned, Disconnect() is called on the user, however Disconnect() is not yet implemented
- Made User.IsRegistered not mapped to DB (because it's unique to each session)

[Trello ticket](https://trello.com/c/m6C5uvUe/87-find-existing-users-in-the-db-on-connect)

### CODE REVIEW

File: Backend/Models/Users/User.cs [hunk 1]
```diff
@@ -56,7 +57,7 @@ public bool IsBanned
         private User()
         {
             // This is needed by EntityFramework
-            _socket = new WebSocket("");
+            _socket = new WebSocket("ws://localhost");
```
Comment (ahmed-okkio) [inline Backend/Models/Users/User.cs:60]: Could we make it nullable since its not mapped anyways?

Comment (OmarQurashi868) [reply Backend/Models/Users/User.cs:60]: We won't have a user object without a socket though

File: Backend/Models/Users/UserManager.cs [hunk 2]
```diff
@@ -72,9 +72,22 @@ public static bool Authenticate(WebSocket socket, string username)
...
+            else
+            {
+                user = userInDb;
+            }
             user.IsRegistered = true;
-            user.SaveToDb();
+
+            if (user.IsBanned)
+            {
+                Disconnect(user);
```
Comment (ahmed-okkio) [inline Backend/Models/Users/UserManager.cs:88]: We should probably return false after this disconnect and log that the user is banned but I guess this is part of the banned work?

File: Backend/Models/Users/UserManager.cs [hunk 3]
```diff
@@ -72,9 +72,22 @@ public static bool Authenticate(WebSocket socket, string username)
...
+            if (userInDb == null)
+            {
+                user.Username = username;
+                user.SaveToDb();
+            }
+            else
+            {
+                user = userInDb;
+            }
             user.IsRegistered = true;
```
Comment (ahmed-okkio) [inline Backend/Models/Users/UserManager.cs]: potentially move this below the isBanned check?

File: Backend/Models/Users/UserManager.cs (hunk 2 above)
Comment (OmarQurashi868) [reply Backend/Models/Users/UserManager.cs:88]: nah you're right, ill get on with it

File: Backend/Models/Users/UserManager.cs (hunk 3 above)
Comment (OmarQurashi868) [reply Backend/Models/Users/UserManager.cs]: will do

File: Backend/Models/Users/User.cs (hunk 1 above)
Comment (ahmed-okkio) [reply Backend/Models/Users/User.cs:60]: Yeah mainly meant to get rid of this from the constructor

Comment (OmarQurashi868) [reply Backend/Models/Users/User.cs:60]: We would then need to have multiple null checks for the socket, where logically we wouldn't have a user without a socket

**Review by ahmed-okkio — APPROVED** [review]

**Pull request merged**